src/Microsoft.Diagnostics.Monitoring.WebApi/Exceptions/ExceptionsConfiguration.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi.Exceptions
+{
+    public class ExceptionsConfiguration
+    {
+        /// <summary>
+        /// The list of exception configurations that determine which exceptions should be shown.
+        /// Each configuration is a logical OR, so if any of the configurations match, the exception is shown.
+        /// </summary>
+        [JsonPropertyName("include")]
+        public List<ExceptionConfiguration> Include { get; set; } = new();
+
+        /// <summary>
+        /// The list of exception configurations that determine which exceptions should be shown.
+        /// Each configuration is a logical OR, so if any of the configurations match, the exception isn't shown.
+        /// </summary>
+        [JsonPropertyName("exclude")]
+        public List<ExceptionConfiguration> Exclude { get; set; } = new();
+
+        /// <summary>
+        /// By default, filters will check for a namespace.name match or a name match for ease-of-use.
+        /// This toggle allows users to limit the filter to only check for an exact namespace.name match.
+        /// </summary>
+        [JsonPropertyName("allowSimplifiedNames")]
+        public bool AllowSimplifiedNames { get; set; } = true;
+
+        private static bool CheckConfiguration(IExceptionInstance exception, List<ExceptionConfiguration> filterList, Func<ExceptionConfiguration, CallStackFrame, bool> evaluateFilterList)
+        {
+            var topFrame = exception.CallStack.Frames.Any() ? exception.CallStack.Frames.First() : null;
+
+            foreach (var configuration in filterList)
+            {
+                if (evaluateFilterList(configuration, topFrame))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        internal bool ShouldInclude(IExceptionInstance exception)
+        {
+            Func<ExceptionConfiguration, CallStackFrame, bool> evaluateFilterList = (configuration, topFrame) =>
+            {
+                bool include = true;
+                if (topFrame != null)
+                {
+                    CompareIncludeValues(configuration.MethodName, GetSimplifiedName(topFrame.MethodName), topFrame.MethodName, ref include);
```

Comment:
Rather than always computing the simplified name, can you do something like `AllowSimplifiedNames ? GetSimplifiedName(topFrame.MethodName) ? topFrame.MethodName` and not have the called methods care about switching between the two values? This should speed up execution time and minimize allocations when they aren't used.

Reply:
I see that there is some kind of fallback where if `AllowSimplifiedNames` is true and the value doesn't match the simple name, then it falls back to checking the full name.